ingest/expression_files/dense_ingestor.py

```diff
@@ -0,0 +1,179 @@
+
+    # @trace
+    # def preprocess(self):
+    #     """Determines if file is R-formatted. Creates dataframe (df)"""
+    #     csv_file, open_file_object = self.open_file(self.file_path)
+    #     dtypes = {'GENE': object}
+    #
+    #     # # Remove white spaces and quotes
+    #     header = [col_name.strip().strip('\"') for col_name in self.header]
+    #     # See if R formatted file
+    #     if (header[-1] == '') and (header[0].upper() != 'GENE'):
+    #         header.insert(0, 'GENE')
+    #         # Although the last column in the header is empty, python treats it
+    #         # as an empty string,[ ..., ""]
+    #         header = header[0:-1]
+    #     else:
+    #         header[0] = header[0].upper()
+    #     # Set dtype for expression values to floats
+    #     dtypes.update({cell_name: 'float' for cell_name in header[1:]})
+    #     self.df = self.open_file(
+    #         self.file_path,
+    #         open_as='dataframe',
+    #         names=header,
+    #         skiprows=1,
+    #         dtype=dtypes,
+    #         # chunksize=100000, Save for when we chunk data
+    #     )[0]
+    def transform(self):
+        """Transforms dense matrix into gene data model.
+        """
+        start_time = datetime.datetime.now()
+        self.error_logger.info('Starting run at ' +
+                               str(start_time), extra=self.extra_log_params)
+        print('Starting run at ' + str(start_time))
+        num_processed = 0
+        gene_models = []
+        data_arrays = []
+        for all_cell_model in self.set_data_array_cells(
+                self.header[1:], ObjectId()):
+            data_arrays.append(all_cell_model)
+        # Represents row as an ordered dictionary
+        for row in self.csv_file:
+            numeric_scores = list(
+                map(lambda x: round(float(x.strip().strip("\'")), 3), row[1:])
+            )
+            valid_expression_scores = []
+            cells = []
+
+            for idx, expression_score in enumerate(numeric_scores, 1):
+                if expression_score != 0:
+                    valid_expression_scores.append(expression_score)
+                    cells.append(self.header[idx])
+            gene = row[0]
+            if gene in self.gene_names:
+                raise ValueError(f'Duplicate gene: {gene}')
+            self.gene_names[gene] = True
+            print(f'Transforming gene :{gene}')
+            formatted_gene_name = gene.strip().strip('\"')
+            id = ObjectId()
+            gene_models.append(self.Model(
+                {
+                    'name': formatted_gene_name,
+                    'searchable_name': formatted_gene_name.lower(),
+                    'study_file_id': self.study_file_id,
+                    'study_id': self.study_id,
+                    '_id': id,
+                    'gene_id': self.matrix_params['gene_id']
+                    if 'gene_id' in self.matrix_params
+                    else None,
```

Comment:
It's hard to read multi-line ternaries that are embedded in dictionaries.  Could you extract this into a simple `gene_id` variable?